### PR TITLE
Update Swift Test App for iOS 13 compatibility

### PR DESCRIPTION
**Pull request by timkimadobe**
## Description

This PR updates the test app for iOS 13 compatibility; some newer SwiftUI additions in iOS 14 were rolled back to iOS 13 implementation. This means that the `@main` and `@StateObject` property wrappers are not available, and app is now initialized using the app + scene delegate flow, and `@StateObject` is reverted to `@ObservedObject`.

## Motivation and Context

Allows for testing new ad ID features in iOS 13, while still using SwiftUI for views.

## How Has This Been Tested?

Simulator running iOS 13.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by cacheung — APPROVED** [review]

File: SampleApps/TestApp/ContentView.swift [hunk 1]
```diff
@@ -258,40 +261,40 @@ struct CustomIdentifierView: View {
...
-
+                
                 Button(action: {
                     if extensions.isEdgeIdentityRegistered {
                         MobileCore.unregisterExtension(AEPEdgeIdentity.Identity.self)
                     } else {
                         MobileCore.registerExtension(AEPEdgeIdentity.Identity.self)
                     }
-
+
```
Comment (emdobrin) [inline SampleApps/TestApp/ContentView.swift:290]: Can you run `make format` or are these changes after formatting?

File: SampleApps/TestApp/SceneDelegate.swift [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    var window: UIWindow?
+    
+    func scene(_ scene: UIScene, willConnectTo session: UISceneSession, options connectionOptions: UIScene.ConnectionOptions) {
+//        let window = UIWindow(frame: UIScreen.main.bounds)
+//        window.rootViewController = UIHostingController(rootView: ContentView())
+//        self.window = window
+//        window.makeKeyAndVisible()
+        
+        // Use a UIHostingController as window root view controller.
+        if let windowScene = scene as? UIWindowScene {
```
Comment (emdobrin) [inline SampleApps/TestApp/SceneDelegate.swift:26]: let's remove commented code and simplify the if { } with a guard 

guard let windowScene = (scene as? UIWindowScene) else { return }
window = ...

File: SampleApps/TestApp/TestApp.swift [hunk 3]
```diff
@@ -11,15 +11,15 @@
...
-    // swiftlint:disable weak_delegate
-    @UIApplicationDelegateAdaptor(AppDelegate.self) var appDelegate
-
-    var body: some Scene {
-        WindowGroup {
-            ContentView()
-        }
-    }
-}
+// Implementation updated to use App + Scene Delegates for iOS 13 compatibility
```
Comment (emdobrin) [inline SampleApps/TestApp/TestApp.swift:14]: let's remove this file if no longer used

File: SampleApps/TestApp/ContentView.swift (hunk 1 above)
Comment (timkimadobe) [reply SampleApps/TestApp/ContentView.swift]: Thank you, forgot to run the autocorrect! Should be applied with the latest commit

File: SampleApps/TestApp/SceneDelegate.swift (hunk 2 above)
Comment (timkimadobe) [reply SampleApps/TestApp/SceneDelegate.swift]: Updated to guard statement!

File: SampleApps/TestApp/TestApp.swift (hunk 3 above)
Comment (timkimadobe) [reply SampleApps/TestApp/TestApp.swift]: Removed file

**Review by timkimadobe — COMMENTED** [review]
Thanks for the review @emdobrin! Applied lint autocorrect and applied suggested changes

**Review by emdobrin — APPROVED** [review]

**Pull request merged**